packages/editor/src/components/provider/index.js

```diff
@@ -223,6 +225,9 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 			setRenderingMode,
 		} = unlock( useDispatch( editorStore ) );
 		const { createWarningNotice } = useDispatch( noticesStore );
+		const { registerBlockBindingsSource } = unlock(
+			useDispatch( blocksStore )
+		);
```

Comment:
That would make sense. I'll try to move it to the `edit-site` and `edit-post` packages. Something I'm wondering is how external developers would hook into that once the APIs are public, but I can take a look at that as well.

Thanks a lot for the guidance!

Reply:
I've changed the approach trying to simulate something similar to how I understood block registration works. I've also updated the description of the pull request to reflect that. Basically, what I did:

Trying to follow a similar approach to how blocks are registered, I did:

* Create a new `registerBlockBindingsSource` function exported by `@wordpress/blocks`. There, I added some validation similar to the one we have in the server: [link](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-block-bindings-registry.php#L98-L183).
* Create a new `registerCoreBlockBindingsSources` function to register the current core sources, post meta and pattern overrides. This function uses the previous one and it is placed in `@wordpress/editor` because some sources, like post meta, might need to access the editor store.
* Call the `registerCoreBlockBindingsSources` from `edit-site` and `edit-post` as suggested.

The code should be simplified once the APIs are public and we should add more functions like `unregisterBlockBindingsSource`, but I believe that could be handled in another pull request.

Please, let me know if that's not what you had in mind.